From 3ced6a0963db3a28e36a044d373b9bcab4589cf1 Mon Sep 17 00:00:00 2001
From: Fugang Duan <fugang.duan@nxp.com>
Date: Mon, 19 Jun 2017 17:10:09 +0800
Subject: [PATCH 1950/5242] MLK-15117 ARM64: dts: imx8qxp-lpddr4-arm2: add
 extended dts for enet2 port

commit  af0623ca6612302009dd81f4d31e7054c8ebabba from
https://source.codeaurora.org/external/imx/linux-imx.git

Since lpddr4 arm2 baord enet2 has pin conflict with esai0,
so add extended dts for enet2 port.

Signed-off-by: Fugang Duan <fugang.duan@nxp.com>
Reviewed-by: Robin Gong <yibin.gong@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/Makefile             |    9 +++++--
 .../freescale/fsl-imx8qxp-lpddr4-arm2-enet2.dts    |   27 ++++++++++++++++++++
 .../boot/dts/freescale/fsl-imx8qxp-lpddr4-arm2.dts |    2 +-
 3 files changed, 35 insertions(+), 3 deletions(-)
 create mode 100644 arch/arm64/boot/dts/freescale/fsl-imx8qxp-lpddr4-arm2-enet2.dts

diff --git a/arch/arm64/boot/dts/freescale/Makefile b/arch/arm64/boot/dts/freescale/Makefile
index 47dc5ef..9369cfc 100644
--- a/arch/arm64/boot/dts/freescale/Makefile
+++ b/arch/arm64/boot/dts/freescale/Makefile
@@ -13,5 +13,10 @@ dtb-$(CONFIG_ARCH_LAYERSCAPE) += fsl-ls2080a-rdb.dtb
 dtb-$(CONFIG_ARCH_LAYERSCAPE) += fsl-ls2080a-simu.dtb
 dtb-$(CONFIG_ARCH_LAYERSCAPE) += fsl-ls2088a-qds.dtb
 dtb-$(CONFIG_ARCH_LAYERSCAPE) += fsl-ls2088a-rdb.dtb
-dtb-$(CONFIG_ARCH_FSL_IMX8QM) += fsl-imx8qm-lpddr4-arm2.dtb fsl-imx8qm-lpddr4-arm2_ca53.dtb fsl-imx8qm-lpddr4-arm2_ca72.dtb fsl-imx8qm-lpddr4-arm2-it6263.dtb
-dtb-$(CONFIG_ARCH_FSL_IMX8QXP) += fsl-imx8qxp-lpddr4-arm2.dtb fsl-imx8qxp-lpddr4-arm2-it6263.dtb
+dtb-$(CONFIG_ARCH_FSL_IMX8QM) += fsl-imx8qm-lpddr4-arm2.dtb \
+				 fsl-imx8qm-lpddr4-arm2_ca53.dtb \
+				 fsl-imx8qm-lpddr4-arm2_ca72.dtb \
+				 fsl-imx8qm-lpddr4-arm2-it6263.dtb
+dtb-$(CONFIG_ARCH_FSL_IMX8QXP) += fsl-imx8qxp-lpddr4-arm2.dtb \
+				  fsl-imx8qxp-lpddr4-arm2-enet2.dtb \
+				  fsl-imx8qxp-lpddr4-arm2-it6263.dtb
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-lpddr4-arm2-enet2.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-lpddr4-arm2-enet2.dts
new file mode 100644
index 0000000..7a7c67f
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-lpddr4-arm2-enet2.dts
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2017 NXP
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+#include "fsl-imx8qxp-lpddr4-arm2.dts"
+
+&esai0 {
+	status = "disabled";
+};
+
+&fec2 {
+	status = "okay";
+};
+
+&ethphy1 {
+	status = "okay";
+};
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-lpddr4-arm2.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-lpddr4-arm2.dts
index 61b0a5b..7c374f3 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-lpddr4-arm2.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-lpddr4-arm2.dts
@@ -334,7 +334,7 @@
 	phy-handle = <&ethphy1>;
 	fsl,ar8031-phy-fixup;
 	fsl,magic-packet;
-	status = "okay";
+	status = "disabled";
 };
 
 &i2c0_csi0 {
-- 
1.7.9.5

